### PR TITLE
Fixed tests and added equals

### PR DESCRIPTION
**Pull request by gustonem**

### CODE REVIEW

File: lost-and-found-service-persistence/src/main/java/cz/muni/fi/dao/CategoryDaoImpl.java [hunk 1]
```diff
@@ -11,11 +13,14 @@
...
 public class CategoryDaoImpl implements CategoryDao {
 
-    @PersistenceContext(unitName = "category-unit", type = PersistenceContextType.EXTENDED)
+    @PersistenceContext
     private EntityManager em;
 
+    @Override
     public void addCategory(Category category) throws IllegalArgumentException {
         if (category == null) {
             throw new IllegalArgumentException("Category");
```
Comment (TerkaSlaninakova) [inline lost-and-found-service-persistence/src/main/java/cz/muni/fi/dao/CategoryDaoImpl.java]: Can you add an if for when category.getId() != null to throw an exception? Or is there a use case when we don't want to do that?

File: lost-and-found-service-persistence/src/test/java/cz/muni/fi/CategoryDaoImplTest.java [hunk 2]
```diff
@@ -2,129 +2,115 @@
...
-        assertEquals(updatedCategory.getName(), newName);
     }
 
     @Test
-    public void shouldReturn0CategoriesWhenEmpty() {
+    public void testGetAllCategoriesWhenEmpty() {
         assertEquals(categoryDao.getAllCategories().size(), 0);
-        assertNull(categoryDao.getCategoryById(0L));
     }
-*/
```
Comment (TerkaSlaninakova) [inline lost-and-found-service-persistence/src/test/java/cz/muni/fi/CategoryDaoImplTest.java]: Can you add test for getAllCategories() when categoryDao is not empty? This scenario is missing.

File: lost-and-found-service-persistence/src/test/java/cz/muni/fi/CategoryDaoImplTest.java [hunk 3]
```diff
@@ -2,129 +2,115 @@
...
     @Test
-    public void shouldUpdateCategory() {
-        categoryDao.addCategory(electro);
-        String newName = "Electronics";
-
-        electro.setName(newName);
-        categoryDao.updateCategory(electro);
-
-        Category updatedCategory = categoryDao.getCategoryById(electro.getId());
+    public void testUpdateCategory() {
```
Comment (TerkaSlaninakova) [inline lost-and-found-service-persistence/src/test/java/cz/muni/fi/CategoryDaoImplTest.java]: Can you modify the test by by adding setAttribute too and make use of getters (getAttribute and getName) in verification?

File: lost-and-found-service-persistence/src/test/java/cz/muni/fi/LocationDaoImplTest.java [hunk 4]
```diff
@@ -5,181 +5,118 @@
...
-        location.setDescription(newDescription);
+        modified.setDescription(newDescription);
 
         locationDao.updateLocation(location);
 
-        Location updatedLocation = locationDao.getLocationById(location.getId());
-
-        assertEquals(updatedLocation.getDescription(), newDescription);
-        assertEquals(updatedLocation.getItems().size(), 0);
+        assertEquals(location, entityManager.find(Location.class, location.getId()));
```
Comment (TerkaSlaninakova) [inline lost-and-found-service-persistence/src/test/java/cz/muni/fi/LocationDaoImplTest.java]: can you add check with getDescription() ?

File: lost-and-found-service-persistence/src/main/java/cz/muni/fi/dao/LocationDaoImpl.java [hunk 5]
```diff
@@ -12,11 +14,14 @@
...
+@Repository
+@Transactional
 public class LocationDaoImpl implements LocationDao {
 
-    @PersistenceContext(unitName = "location-unit", type = PersistenceContextType.EXTENDED)
+    @PersistenceContext
     private EntityManager em;
 
+    @Override
     public void addLocation(Location location) throws IllegalArgumentException {
```
Comment (TerkaSlaninakova) [inline lost-and-found-service-persistence/src/main/java/cz/muni/fi/dao/LocationDaoImpl.java]: Can you add an if for when location.getId() != null to throw an exception? Or is there a use case when we don't want to do that?

File: lost-and-found-service-persistence/src/main/java/cz/muni/fi/dao/CategoryDaoImpl.java (hunk 1 above)
Comment (jkbpolacek) [reply lost-and-found-service-persistence/src/main/java/cz/muni/fi/dao/CategoryDaoImpl.java]: fixol som ja

File: lost-and-found-service-persistence/src/main/java/cz/muni/fi/dao/LocationDaoImpl.java (hunk 5 above)
Comment (gustonem) [reply lost-and-found-service-persistence/src/main/java/cz/muni/fi/dao/LocationDaoImpl.java]: Jakub to spravil.
